### PR TITLE
DPC-1595 Run unit tests for v2 in GHA

### PR DESCRIPTION
**Pull request by MrBilnon**
### Fixes [DPC-1595](https://jira.cms.gov/browse/DPC-1595)
We need to be able to run v2 linting and unit tests in GHA to verify tests work.

### Proposed Changes
For API v2 and Attribution v2:
* Add container-based lint and unit testing
* Add GHA jobs

### Change Details
* Add Make target for container-based testing 
* Move linting from separate Make target to `unit_test.sh`
* Make edits required to pass linting
* Test GHA by introducing linting and unit test errors

### Security Implications
- [ ] new software dependencies
- [x] security controls or supporting software altered
GitHub now requires these two new GHA jobs to succeed before merging
Screen Shot 2021-08-29 at 5 48 22 PM

- [ ] new data stored or transmitted
An encrypted value is added in this PR for testing BFD hashing. This is not a new data element for DPC.

- [x] security checklist is completed for this change
Applies to "New code; scans" section of [this SIA](https://confluence.cms.gov/pages/viewpage.action?pageId=490380420)

- [ ] requires more information or team discussion to evaluate security implications
- [x] no PHI/PII is affected by this change

#### Lint errors fail GHA job
Screen Shot 2021-08-29 at 4 56 19 PM

#### Unit test errors fail GHA jobs
Screen Shot 2021-08-29 at 4 31 17 PM

#### New GHA jobs are successful when lint and unit tests are clean
Screen Shot 2021-08-29 at 4 43 35 PM

### Feedback Requested
* Are there other approaches that would work better?

### CODE REVIEW

File: dpc-go/dpc-api/src/v2/patient.go [hunk 1]
```diff
@@ -52,9 +52,7 @@ func (pc *PatientController) Export(w http.ResponseWriter, r *http.Request) {
 
 	bf := make([]model.BatchFile, 0)
 	for _, b := range batches {
-		for _, f := range *b.Files {
-			bf = append(bf, f)
-		}
+		bf = append(bf, *b.Files...)
```
Comment (MrMorie) [inline dpc-go/dpc-api/src/v2/patient.go:55]: 😎

File: dpc-go/dpc-attribution/unit_test.sh [hunk 2]
```diff
@@ -2,15 +2,25 @@
...
+set -e
+set -o pipefail
+
 timestamp=`date +%Y-%m-%d_%H-%M-%S`
 mkdir -p test_results/${timestamp}
 mkdir -p test_results/latest
 
-echo "Running GO DPC-ATTRIBUTION unit tests and placing them in test_results/${timestamp}..."
 cd src
+echo -e "-------------- DPC-API LINTING STARTED --------------"
```
Comment (MrMorie) [inline dpc-go/dpc-attribution/unit_test.sh:13]:
```suggestion
echo -e "-------------- DPC-ATTRIBUTION LINTING STARTED --------------"
```

File: dpc-go/dpc-attribution/unit_test.sh [hunk 3]
```diff
@@ -2,15 +2,25 @@
...
+golangci-lint --timeout 5m run echo "*********** DPC-API IS LINT FREE!! ***********" || echo -e "*********** LINTING FAILED!! ***********"
```
Comment (MrMorie) [inline dpc-go/dpc-attribution/unit_test.sh:14]:
```suggestion
golangci-lint --timeout 5m run echo "*********** DPC-ATTRIBUTION IS LINT FREE!! ***********" || echo -e "*********** LINTING FAILED!! ***********"
```

**Review by MrMorie — CHANGES_REQUESTED** [review]
LGTM, just needs the copy/pasta typo fixed.

Also, dpc-go/dpc-attribution/ops/scripts/secrets seems to be a new file, but it is empty. Was that intended to be committed?

**Review by MrMorie — DISMISSED** [review]
LGTM

**Comment by dhgreene** [comment]
> LGTM, just needs the copy/pasta typo fixed.
> 
> Also, dpc-go/dpc-attribution/ops/scripts/secrets seems to be a new file, but it is empty. Was that intended to be committed?

Yes--it gets past an error in some cases.

**Comment by MrMorie** [comment]
> > LGTM, just needs the copy/pasta typo fixed.
> > Also, dpc-go/dpc-attribution/ops/scripts/secrets seems to be a new file, but it is empty. Was that intended to be committed?
> 
> Yes--it gets past an error in some cases.

👍

**Review by MrMorie — APPROVED** [review]
:shipit:

**Review by jonfulk — APPROVED** [review]
🧼

**Pull request merged**